exec/testing/fake_exec.go

```diff
@@ -28,13 +33,34 @@ type FakeExec struct {
 	CommandScript []FakeCommandAction
 	CommandCalls  int
 	LookPathFunc  func(string) (string, error)
+
+	T *testing.T
+}
+
+// NewFakeExec creates a FakeExec. You can pass nil for t if you are not running from a
+// test program. (This will cause certain failures to result in a panic rather than a
+// t.Fatal.) cmdsInPath is a list of command names for which LookPath will return a match
+// ("/fake-bin/commandname"); if your code does not use LookPath you can pass nil. (If you
+// need more complex behavior, you can override LookPathFunc on the returned FakeExec.)
+func NewFakeExec(t *testing.T, cmdsInPath []string) *FakeExec {
```

Comment:
> Why would I call this function in the first place if I want to change LookPathFunc, considering that's pretty much the only added value of this function?

I guess I like having New methods rather than manually constructing somebody-else's-struct-type by hand. I had added `func NewFakeExec(t *testing.T) *FakeExec` before adding the `cmdsInPath` arg.

But yes, I could do it the way you suggested instead.